docs/marketplace/add_to_marketplace/declarative_marketplace.md

```diff
@@ -0,0 +1,268 @@
+
+:::tip
+
+We suggest to mask the variable and to protect it, as suggested in [Gitlab documentation](https://docs.gitlab.com/ee/ci/variables/#cicd-variable-security).
+
+:::
+
+#### Setup the pipeline file
+
+Create a `.gitlab-ci.yml` file with the following contents:
+
+```yaml
+stages:
+  - preparation
+  - apply-marketplace
+
+variables:
+  # make sure to use the latest miactl version available https://github.com/mia-platform/miactl/releases
+  MIACTL_VERSION: "0.8.0"  
+
+preparation:
+  stage: preparation
+  image: alpine:latest
+  script:
+    - apk add --no-cache git
+    - git clone $CI_REPOSITORY_URL ./my-repo
+    - cd ./my-repo
+
+apply-marketplace:
+  stage: apply-marketplace
+  image: ghcr.io/mia-platform/miactl:$MIACTL_VERSION
+  script:
+    - 'echo "COMPANY_ID: $COMPANY_ID"'
+    - 'echo "CONSOLE_ENDPOINT: $CONSOLE_ENDPOINT"'
+    - miactl context auth new-auth --client-id $CLIENT_ID --client-secret $CLIENT_SECRET
+    - miactl context set new-context --company-id $COMPANY_ID --endpoint $CONSOLE_ENDPOINT --auth-name new-auth
+    - miactl context use new-context
+    - miactl marketplace apply -f ./resources
+```
+
+:::caution
+
+This pipeline is just an example. You may need to implement additional checks and use different images.
+
+:::
+
+Here is an explanation of the environment variables needed in the pipeline:
+
+- `MIACTL_VERSION`: the version of miactl, make sure to always use the latest version in order to benefit the latest features and bug fixes.
+- `CLIENT_ID`: a secret env containing the Service Account Client ID.
+- `CLIENT_SECRET`: a secret env containing the Service Account Client Secret.
+- `CONSOLE_ENDPOINT`: the base url of your Console installation.
+- `COMPANY_ID`: the Company ID where you want the Marketplace Items to be created in.
+
+## An example of interaction with the repository
+
+Acme Corporation uses the Console as its Internal Development Platform.
+
+Developers of the Company can open Merge Requests to the Marketplace Git repository created above, and Maintainers can approve and merge the MRs.
+
+Imagine that a developer of Acme Corporation creates the Inventory System plugin, as in the example above.
+They now want to add this plugin to their company. 
+
+### The Merge Request process
+
+In order to do so, they can open a Merge Request to the Gitlab Repository created above: 
+
+![Declarative-CJ-Merge-Request](img/declarative_mr.png)
+
+Once the MR is merged, the pipeline start applying the items in the repository; among them we can see that the Inventory System has been inserted successfully:
+
+![Declarative-CJ-Pipeline-Log](img/declarative_pipeline_log.png)
+
+
+Let's now assume that a developer spots a mistake in the plugin description. Similarly, they can open a Merge Request that addresses the error:
+
+![Declarative-CJ-fix-MR](img/declarative_fix_mr.png)
+
+After a maintainer merges the changes, a new pipeline is run, this time resulting in an Update of the item:
```

Comment:
After a maintainer merges the changes, a new pipeline is triggered, leading to an update of the item.

Reply:
```suggestion
After a maintainer merges the changes, a new pipeline is triggered, leading to an update of the item.
```